src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VALUE = 10;
+    private static final double MORE_SIZE = 1.5;
+
+    private Object[] array;
+
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_VALUE];
```

Comment:
in such way you won't need to cast objects to T in methods
```suggestion
        array = (T[]) new Object[DEFAULT_VALUE];
```